packages/block-library/src/gallery/save.js

```diff
@@ -46,6 +46,22 @@ export default function save( { attributes } ) {
 					</li>
 				);
 			} ) }
+		</>
+	);
+
+	if ( galleryCaption ) {
```

Comment:
If there is no change in the markup for the case when we don't provide the caption, do we really need to build all deprecation story? It seems like it should out of the box if the caption is not provided. I don't see any changes in the deprecated version serialized in the fixtures folder (input is almost identical to the output when you ignore whitespaces).

Reply:
I'm not sure! The markup of the new version is the same but there's an added attribute. Happy to remove the deprecation if it's not needed though 🙂

Reply:
You would have to tests it but I know we had similar cases in the past and we didn't have to build deprecation logic when new optional HTML attributes were added or wrapped with optional HTML tags. It seems like this is the case here.